### ISSUE TITLE
Seemingly dangling type reference to aRN:ARN

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-aws#1682]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

Currently `schema.json` contains the following construcs:

```
      "$ref": "#/types/aws:index/aRN:ARN",
```

But there is no corresponding entry in `.types`, so this is a dangling reference. In addition these constructs happen to be used in addition to `"type": "string"` which seems suspect.

### Steps to reproduce

```
  jq '.resources["aws:datasync/efsLocation:EfsLocation"]' provider/cmd/pulumi-resource-aws/schema.json

  ...
  
  "inputProperties": {
    "efsFileSystemArn": {
      "type": "string",
      "$ref": "#/types/aws:index/aRN:ARN",
      "description": "Amazon Resource Name (ARN) of EFS File System.\n"
    },

```

Expected: no dangling ref, either string or ref
Actual: string and dangling ref

**Comment by stack72** [comment]
@t0yv0 this was added by @pgavlin in https://github.com/pulumi/pulumi-aws/pull/991

via the following code:

```
			"aws_datasync_location_efs": {
				Tok: awsResource(datasyncMod, "EfsLocation"),
				Fields: map[string]*tfbridge.SchemaInfo{
					"efs_file_system_arn": {Type: awsTypeDefaultFile(awsMod, "ARN")},
				},
			},
```

Notice `awsTypeDefaultFile` which does the following:

```
func awsTypeDefaultFile(mod string, typ string) tokens.Type {
	fn := string(unicode.ToLower(rune(typ[0]))) + typ[1:]
	return awsType(mod, fn, typ)
}
```

This means we could have been carrying this for quite a while

thoughts?

**Comment by t0yv0** [comment]
Thanks for looking that up! Yes, it might be intended. I do not think it is doing any real harm now just odd corner cases here and there. I stumbled on it when working on programgen for $fn.Output forms, but eventually concluded it's a broader unrelated issue with dangling imports being generated:

https://github.com/pulumi/pulumi/issues/8324

Then I got a workaround in my PR so I can proceed.

For I just would like to clarify and document intent behind this aRN:ARN. I would also like to check why `pulumi schema check` accepts this. It seems like schema should not validate, and dangling references should be made prominent.
